lib/spack/spack/report.py

```diff
@@ -272,3 +280,13 @@ def test_context_manager(
             yield
     finally:
         reporter.test_report(filename, specs=collector.specs)
+
+
+def default_log_file(spec):
+    """Computes the default filename for the log file and creates
+    the corresponding directory if not present
+    """
+    basename = spec.format_path("test-{name}-{version}-{hash}.xml")
```

Comment:
IMO The name of the directory should be tied to the context.  So only stand-alone tests should have files/directories with names starting `test` and build should use `build`.